src/formElementPlugins/listBuilder/src/ApplicationView.tsx

```diff
@@ -0,0 +1,347 @@
+              onClick={updateList}
+            />
+            {displayType === 'table' && selectedListItem !== null && (
+              <Button
+                secondary
+                content={deleteItemText}
+                onClick={() => deleteItem(selectedListItem)}
+              />
+            )}
+            {inputError && <p>{strings.ERROR_LIST_ITEMS_NOT_VALID}</p>}
+          </Form>
+        </Segment>
+      </Modal>
+      {DisplayComponent}
+    </>
+  )
+}
+
+export default ApplicationView
+
+const buildElement = (field: TemplateElement, index: number) => ({
+  id: index, // This is the only link between element and the response
+  code: field.code,
+  pluginCode: field.elementTypePluginCode as string,
+  category: field.category as TemplateElementCategory,
+  title: field.title as string,
+  parameters: field.parameters,
+  validationExpression: field?.validation || true,
+  validationMessage: field?.validationMessage || '',
+  isRequired: field.isRequired ?? true,
+  // Hard-coded visisbility and editability (for now)
+  isEditable: true,
+  isVisible: true,
+  // "Dummy" values, but required for element props:
+  elementIndex: 0,
+  page: 0,
+  sectionIndex: 0,
+  helpText: null,
+  sectionCode: '0',
+})
+
+const getDefaultDisplayFormat = (inputFields: TemplateElement[]) => {
+  const displayString = inputFields.reduce(
+    (acc: string, { code, title }) => acc + `**${title}**: \${${code}}  \n`,
+    ''
+  )
+  return { header: '', meta: '', description: displayString }
+}
+
+const resetCurrentResponses = (inputFields: TemplateElement[]) =>
+  new Array(inputFields.length).fill({ value: { text: undefined } })
+
+const buildListItem = (item: InputResponseField[], inputFields: TemplateElement[]) =>
+  item.reduce(
+    (acc, field, index) => ({
+      ...acc,
+      [inputFields[index].code]: {
+        code: inputFields[index].code,
+        id: field.id,
+        title: inputFields[index].title,
+        value: field.value,
+      },
+    }),
+    {}
+  )
+
+const getInputResponses = (listItem: ListItem, inputFields: TemplateElement[]) =>
+  inputFields.reduce((acc: InputResponseField[], { code }) => [...acc, listItem[code]], [])
+
+const anyInvalidItems = (currentInput: InputResponseField[]) =>
+  currentInput.some((response) => response.isValid === false)
+
+const anyIncompleteItems = (currentInput: InputResponseField[], inputFields: TemplateElement[]) =>
+  currentInput.some(
+    (response, index) => inputFields[index]?.isRequired !== false && !response.value.text
+  )
+
+const anyErrorItems = (currentInput: InputResponseField[], inputFields: TemplateElement[]) =>
+  anyInvalidItems(currentInput) || anyIncompleteItems(currentInput, inputFields)
+
+const substituteValues = (parameterisedString: string, item: any) => {
+  const getValueFromCode = (_: string, $: string, code: string) => item[code].value.text || ''
+  return parameterisedString.replace(/(\${)(.*?)(})/gm, getValueFromCode)
```

Comment:
I know I've added a comment about custom card formatting being probably out of scope, but this is too cool.